### PR TITLE
Storage Test Solution 31/10

### PR DESCRIPTION
**Pull request by VadimKadasiev**

### CODE REVIEW

File: src/main/java/core/basesyntax/Storage.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package core.basesyntax;
 
 public interface Storage<K, V> {
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Storage.java:4]: undo this change

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INDEX_KEY = 0;
+    private static final int INDEX_VALUE = 1;
+    private int arraySize = 0;
+    private int arrayIndex = 0;
+    private final Object[][] tempValuesArray = new Object[10][2];
+    private Object[][] resultValuesArray = new Object[0][0];
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:11]: image

**Review by okuzan — CHANGES_REQUESTED** [review]
Interesting approach! Although you don't use K,V type parameters
Other things to consider:
- 0 is the default for int
- Extract common logic to a private method (iterating and comparing keys)
- Do you really need a temp array as a field?
- simplify equals logic -  return (key1 == key2) || (key1 != null && key2 != null && key1.equals(key2));

**Comment by VadimKadasiev** [comment]
Hi! I was looking for a way to rid of the temporary array. Not sure about array size solution.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,49 @@
...
     }
 
     @Override
     public int size() {
-        return -1;
+        return arraySize;
     }
 }
+
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:51]:
```suggestion
```

**Review by kozhukhovsky — APPROVED** [review]
